lib/liquid/nimble_parser.ex

```diff
@@ -1,27 +1,302 @@
 defmodule Liquid.NimbleParser do
+  @moduledoc """
+  Transform a valid liquid markup in an AST to be executed by `render`
+  """
+  # TODO: Find methods to split this module
   import NimbleParsec
-  alias Liquid.Combinators.{
-    General,
-    Expression
-  }
 
-  defparsec(:literal, General.literal())
-  defparsecp(:expression,
+  alias Liquid.Combinators.General
+
+  liquid_variable =
+    General.start_variable()
+    |> concat(parsec(:variable_name))
+    |> concat(General.end_variable())
+    |> tag(:variable)
+    |> optional(parsec(:__parse__))
+
+  defparsec(:liquid_variable, liquid_variable)
+  defparsec(:variable_name, General.variable_name())
+  defparsec(:start_tag, General.start_tag())
+  defparsec(:end_tag, General.end_tag())
+
+  ################################        Tags              ###########################
+
+  # |> concat(value)
+  assign =
+    empty()
+    |> parsec(:start_tag)
+    |> concat(ignore(string("assign")))
+    |> concat(parsec(:variable_name))
+    |> concat(ignore(string("=")))
+    |> concat(parsec(:value))
+    |> concat(parsec(:end_tag))
+    |> tag(:assign)
+    |> optional(parsec(:__parse__))
+
+  defparsec(:assign, assign)
+
+  decrement =
+    empty()
+    |> parsec(:start_tag)
+    |> string("decrement")
+    |> concat(parsec(:variable_name))
+    |> concat(parsec(:end_tag))
+    |> tag(:decrement)
+    |> optional(parsec(:__parse__))
+
+  defparsec(:decrement, decrement)
+
+  increment =
+    empty()
+    |> parsec(:start_tag)
+    |> string("increment")
+    |> concat(parsec(:variable_name))
+    |> concat(parsec(:end_tag))
+    |> tag(:increment)
+    |> optional(parsec(:__parse__))
+
+  defparsec(:increment, increment)
+
+  ################################           raw          ###########################
+
+  raw_tag =
+    string("raw")
+    |> ignore()
+
+  raw_end_tag =
+    string("endraw")
+    |> ignore()
+
+  open_tag_raw = empty() |> parsec(:start_tag) |> concat(raw_tag) |> concat(parsec(:end_tag))
```

Comment:
Can you change these to use multi-line piping to be consistent with the other definitions?